src/types.ts

```diff
@@ -513,6 +513,10 @@ export type GetUnreadCountAPIResponse = APIResponse & {
   total_unread_count: number;
 };
 
+export type GetUnreadCountBatchAPIResponse = APIResponse & {
+  counts_by_user: GetUnreadCountAPIResponse[];
```

Comment:
Should this be a map (object), instead of an array? AFAIK, the key of the map is the user id

I think this should work (_not tested_)

```suggestion
  counts_by_user: { [string]: GetUnreadCountAPIResponse };
```

Reply:
good catch, the other parts of this code seems to be using `Record` so maybe
`counts_by_user: Record<string, GetUnreadCountAPIResponse>;`?

Reply:
Yeah probably, but my Typescript-foo isn't strong enough to say for sure 😆 @MartinCupela probably knows

Reply:
Jumping in here... `counts_by_user: Record<string, GetUnreadCountAPIResponse>;` should be good.
But probably it will be better if we are a bit more explicit and have:
`{ [userId: string]: GetUnreadCountAPIResponse }`

Reply:
thanks for the input, pushed a fix for this